t/003_baseosver.t

```diff
@@ -0,0 +1,38 @@
+use strict;
+use warnings;
+use File::Basename;
+
+use Test::More tests => 4;
+
+my $scriptdir = dirname(__FILE__);
+
+my $home = $ENV{HOME};
+
+my $os_id = `cat /etc/os-release`;
+$os_id =~ /^ID="?(\w+)"?/m;
+$os_id = $1;
+
+my $os_seen = `./debbuild.out --showrc`;
+like($os_seen, qr/^%\Q$os_id\E ==> (\S+)/m, "os-release ID present");
+
+my $testdir = $scriptdir . '/../testdir/baseosver';
+local $ENV{HOME} = $testdir;
+open my $f, '>', "$testdir/.debmacros" || die "Could not write test file .debmacros: $!";
+print $f "%$os_id somethingsilly\n";
+close $f;
+
+my $test_os_seen = `./debbuild.out --showrc`;
+$test_os_seen =~ /^%\Q$os_id\E ==> (\S+)/m;
+is($1, "somethingsilly", ".debmacros overwrites /etc/os-release");
+
+TODO: {
+    todo_skip "make different /etc/os-release testable", 2;
```

Comment:
Sorry for our mess. debbuild was not originally designed to be unit-testable, and I'm in the process of adding enough scaffolding that we can have proper unit tests of individual functions, which is likely the best way of testing different os-releases.

Until then, the way this test is structured is the best we've got for now.